### ISSUE TITLE
Loop counter incremented twice per iteration

### ISSUE THREAD

**Issue by danluu** [stan-dev/math#816]
#### Description:

`wishart_rng_test.cpp`:
~~~
  int N = 10000;
  double avg = 0;
  ...
  for (int count = 0; count < N; ++count) {
    a = wishart_rng(5.0, sigma, rng);
    avg += log(determinant(a)) / N;
    count++;
  }
~~~

Unless I'm missing something here, this appears to produce only half of the expected average?

#### Current Version:
v2.17.0

**Comment by bob-carpenter** [comment]
Yup, that looks wrong.  Wonder how it passes the test.  Also, that divide by N should only happen once, not that it's going to be a big bottleneck with double values in a 10K-element loop in a test.

**Comment by danluu** [comment]
> that divide by N should only happen once

Do you mean that it should be hoisted out of the loop and done at the end? I might as well do that and fix the count in a single PR.

**Comment by bob-carpenter** [comment]
I don't think it's a big deal in the tests.  If this was in the actual API code, then I'd want to fix it.

**Comment by danluu** [comment]
Closing because this was fixed by #818.
